src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -171,7 +183,80 @@ def acr_update_set(client,
+        if 'name' not in parameters.storage_account:
+            raise CLIError(
+                "Storage account name is required to update " +
+                "the storage account used by a container registry.")
+        if 'access_key' not in parameters.storage_account:
+            parameters.storage_account['access_key'] = get_access_key_by_storage_account_name(
+                parameters.storage_account['name'])
 
     return client.update(resource_group_name, registry_name, parameters)
+
+
+def acr_login(registry_name, resource_group_name=None, username=None, password=None):
+    """Login to a container registry through Docker.
+    :param str registry_name: The name of container registry
+    :param str resource_group_name: The name of resource group
+    :param str username: The username used to log into the container registry
+    :param str password: The password used to log into the container registry
+    """
+    try:
+        call(["docker", "ps"], stdout=PIPE, stderr=PIPE)
+    except Exception as e:
+        raise CLIError("Please verify whether docker is installed and running properly")
+
+    login_server = get_registry_login_server_by_name(registry_name, resource_group_name)
+
+    # 1. if username was specified, verify that password was also specified
+    if username:
+        if not password:
+            try:
+                password = prompt_pass(msg='Password: ')
+            except NoTTYException:
+                raise CLIError('Please specify both username and password in non-interactive mode.')
+
+    # 2. if we don't yet have credentials, attempt to get a refresh token
+    if not password:
+        try:
+            username = "00000000-0000-0000-0000-000000000000"
+            password = get_login_refresh_token(login_server)
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning("AAD authentication failed with message: %s", str(e))
+
+    # 3. if we still don't have credentials, attempt to get the admin credentials (if enabled)
+    if not password:
+        try:
+            cred = acr_credential_show(registry_name)
+            username = cred.username
+            password = cred.passwords[0].value
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning("Admin user authentication failed with message: %s", str(e))
+
+    # 4. if we still don't have credentials, prompt the user
+    if not password:
+        try:
+            username = prompt('Username: ')
+            password = prompt_pass(msg='Password: ')
+        except NoTTYException:
+            raise CLIError(
+                'Unable to authenticate using AAD or admin login credentials. ' +
+                'Please specify both username and password in non-interactive mode.')
+
+    call(["docker", "login",
+          "--username", username,
+          "--password", password,
+          login_server])
```

Comment:
You will need to run the container with 
```
docker run -it -v /var/run/docker.sock:/var/run/docker.sock azuresdk/azure-cli-python
```
Also `apk add docker` if `docker` is not installed in the container.